coredocument/processor.go

```diff
@@ -93,7 +97,8 @@ func (dp defaultProcessor) Send(ctx *header.ContextHeader, coreDocument *coredoc
 		NetworkIdentifier:  dp.config.GetNetworkID(),
 	}
 
-	resp, err := client.SendAnchoredDocument(ctx.Context(), &p2ppb.AnchorDocumentRequest{Document: coreDocument, Header: p2pheader})
+	c, _ := context.WithTimeout(ctx.Context(), dp.config.GetP2PConnectionTimeout())
```

Comment:
yes but thats only for connection, this timeout is needed for all p2p calls otherwise the it waits a long time when connection is about to disconnect.